src/index.js

```diff
@@ -26,7 +26,8 @@ export default function mitt(all: EventHandlerMap) {
 		 * @memberOf mitt
 		 */
 		on(type: string, handler: EventHandler) {
-			(all[type] || (all[type] = [])).push(handler);
+			var arr = all[type] = all[type] || [];
+			return arr.splice.bind(arr, arr.push(handler) - 1, 1);
```

Comment:
this fails if you remove an earlier-inserted event handler, because it shifts the indices:

```js
let e = mitt(), a=()=>{}, b=()=>{};

e.on('a', a);
e.on('a', b);

e.off('a', a);
e.off('a', b);  // attempts to remove all['a'][1], index is incorrect
```